rstudio/rstudio
# Make workbench tab labels localizable

### Intent

Addresses #10208, making the IDE tab labels localizable (e.g. "Environment", "Console", and so forth).

This does not tackle making the tab names in the Pane Layout section of Global Options localizable; those remain hardcoded in English (using the internal tab identifiers, which in English are the same as the labels, but not true in non-English). This PR is already huge, so will open a separate tracking issue on that (#11061).

### Approach

Use localizable strings for the labels when tabs are being created, and created and used constants for the internal tab name identifiers to make it clearer what's intended for the UI versus internal.

### Automated Tests

We don't have tests for non-English UI, but the existing tests will help confirm nothing has been regressed in English.

### QA Notes

Nothing to test specificially; you can switch into French UI and confirm that tab names are now localized, but mainly we are concerned with ensuring we haven't broken anything in English, and that is adequately covered by existing automation and regular usage.

### Checklist
